shared/controller-api/src/main/java/io/pravega/shared/controller/event/ScaleOpEvent.java

```diff
@@ -9,24 +9,40 @@
  */
 package io.pravega.shared.controller.event;
 
-import lombok.AllArgsConstructor;
 import lombok.Data;
 
 import java.util.AbstractMap;
 import java.util.List;
 import java.util.concurrent.CompletableFuture;
+import java.util.stream.Collectors;
 
 @Data
-@AllArgsConstructor
 public class ScaleOpEvent implements ControllerEvent {
     private static final long serialVersionUID = 1L;
     private final String scope;
     private final String stream;
     private final List<Integer> segmentsToSeal;
-    private final List<AbstractMap.SimpleEntry<Double, Double>> newRanges;
+    private final List<AbstractMap.SimpleEntry<Long, Long>> newRanges;
     private final boolean runOnlyIfStarted;
     private final long scaleTime;
 
+    public ScaleOpEvent(String scope, String stream, List<Integer> segmentsToSeal, List<AbstractMap.SimpleEntry<Double, Double>> newRange, boolean runOnlyIfStarted, long scaleTime) {
+        this.scope = scope;
+        this.stream = stream;
+        this.segmentsToSeal = segmentsToSeal;
+        this.newRanges = newRange.stream()
+                .map(x -> new AbstractMap.SimpleEntry<>(Double.doubleToRawLongBits(x.getKey()), Double.doubleToRawLongBits(x.getValue())))
+                .collect(Collectors.toList());
+        this.runOnlyIfStarted = runOnlyIfStarted;
+        this.scaleTime = scaleTime;
+    }
+
+    public List<AbstractMap.SimpleEntry<Double, Double>> getNewRanges() {
```

Comment:
I don't see this method used or tested anywhere.

Reply:
This looks similar to some code in the constructor. Should we invoke it from there?

Reply:
I thought the issue was because of serialization and deserialization using JavaSerializer causing precision loss. 
but i was wrong. I am able to reproduce it and found the problem source.

Reply:
@andreipaduroiu constructor takes double and converts it to long.
getter converts the field from long back into double and returns it.

So the contract for callers appears to be `double` but the field stores `long`

Reply:
@shiveshr 

> I thought the issue was because of serialization and deserialization using JavaSerializer causing precision loss. I am not sure if that can be an issue though, so this is a safe fix that has been put inhere.
>
> Also, I am able to reproduce the issue and have been able to identify the real source of problem.
The real fix for the problem observed is in AutoScaleTask.java and we have a corresponding unit test added too.

This is not really answering my question, why are we adding a method that we aren't using anywhere?

Reply:
This method It is being used to take scale inputs to start the scale workflow. 
The method signature of the getter has been kept identical to earlier, while the field is changed from a map<double, double> to map<long, long>